xrdsst/controllers/client.py

```diff
@@ -31,6 +36,15 @@ def add_client(self, configuration):
             for client in security_server["clients"]:
                 self.remote_add_client(ss_configuration, client)
 
+    def add_service_description(self, configuration):
+        self.init_logging(configuration)
+        for security_server in configuration["security_server"]:
+            BaseController.log_info('Starting service description add process for security server: ' + security_server['name'])
+            ss_configuration = self.initialize_basic_config_values(security_server, configuration)
+            for client in security_server["clients"]:
+                for service_description in client["service_descriptions"]:
```

Comment:
I think this should be conditional, there might well be no service descriptions defined yet in which case this fails...

Reply:
1. What is the point of calling add_service_description() with none defined in the configuration file? 2. This is a for loop, which runs when there are some values present.

Reply:
But it fails here with KeyError then, something that later is tried for amending at ``remote_add_service_description``, but without effect due to the call site already requiring it. If anywhere, this is the place for conditional here.

Reply:
Hmm, I wrote previous comment while multitasking, so I am not certain I made the correct code reference. However, simply put, this fails whenever client does not have ``service_descriptions``:

## Config section
```
  clients:
    - member_class: GOV
      member_code: 1234
      subsystem_code: OMNIBUS
      connection_type: HTTP
```

## Result

```
xrdsst -c config/ss5.yaml   client   add-service-description 
Starting service description add process for security server: ss5
API key for security server: ss5 has already been created
Traceback (most recent call last):
  File "/home/user/xrdsst-review/X-Road-Security-Server-toolkit/env/bin/xrdsst", line 11, in <module>
    load_entry_point('xrdsst', 'console_scripts', 'xrdsst')()
  File "/home/user/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/main.py", line 154, in main
    app.run()
  File "/home/user/xrdsst-review/X-Road-Security-Server-toolkit/env/lib/python3.8/site-packages/cement/core/foundation.py", line 916, in run
    return_val = self.controller._dispatch()
  File "/home/user/xrdsst-review/X-Road-Security-Server-toolkit/env/lib/python3.8/site-packages/cement/ext/ext_argparse.py", line 808, in _dispatch
    return func()
  File "/home/user/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/controllers/client.py", line 25, in add_description
    self.add_service_description(self.load_config())
  File "/home/user/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/controllers/client.py", line 45, in add_service_description
    for service_description in client["service_descriptions"]:
KeyError: 'service_descriptions'
```